tests/Custom/SessionResetTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Mink\WebdriverClassDriver\Tests\Custom;
+
+use Behat\Mink\Tests\Driver\TestCase;
+
+class SessionResetTest extends TestCase
+{
+    /**
+     * @dataProvider initialWindowNameDataProvider
+     */
+    public function testSessionResetClosesWindows(?string $initialWindowName): void
+    {
+        $session = $this->getSession();
+        $session->visit($this->pathTo('/window.html'));
+
+        if (null !== $initialWindowName) {
+            $session->executeScript('window.name = "'.$initialWindowName.'";');
+        }
+
+        $page = $session->getPage();
+
+        $page->clickLink('Popup #1');
+        $page->clickLink('Popup #2');
+
+        $expectedInitialWindowName = $session->evaluateScript('window.name');
+
+        $windowNames = $session->getWindowNames();
+        $this->assertCount(3, $windowNames);
+
+        $session->reset();
+
+        $windowNames = $session->getWindowNames();
+        $this->assertCount(1, $windowNames);
```

Comment:
I'm not fully sure I understand the problem being solved here.

Why are you expecting 1 window after a session reset?

Reply:
Because the `WebdriverClassicDriver::reset` method closes all opened windows, except a window, that opened upon session start.

I haven't changed this behavior (I like that). I've added a test to ensure it's working as expected.

Reply:
OK, so to be clear, the correct behaviour is to keep the initial window open?

Should the window refresh/reload? Otherwise it's possible it "contaminates" the session after reset (e.g. if the window contains some web app that saves session data as cookies/local-storage periodically). 🤔

Reply:
> OK, so to be clear, the correct behaviour is to keep the initial window open?

Yes, that is how you implemented it. Do you remember this? There was no similar code in the MinkSelenim2Driver.

> Should the window refresh/reload? Otherwise it's possible it "contaminates" the session after reset (e.g. if the window contains some web app that saves session data as cookies/local-storage periodically). 🤔

The `reset` method clears cookies and closes any windows opened by the test. It's expected, that the next test will visit the desired page and since the cookies are absent everything will reset by itself (this happens in the driver test suite).

Reply:
> Do you remember this?

Nope. 😅 I do remember that the windows had to be closed at some point, but I really thought that was the same behaviour across all drivers.

> It's expected, that the next test will visit the desired page

Ok, sounds reasonable. 👍